### DISCUSSION TITLE
Optimal parameters for cogagent-vqa - do_sample? top_p? top_k? temperature?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [THUDM/CogVLM#314] [Q&A]
Hello. I am working on a tutorial for using cogagent-vqa for image captioning for Stable Diffusion training.

Could you let me know optimal parameters for captioning images? it will be 1 question and answer

Also is cogagent-vqa best for this?
 do_sample : false or true?
 top_p : what value?
 top_k : what value?
 temperature : what value?

can you tell me default values to set them?

**Comment by zRzRzRzRzRzRzR** [comment]
For the default parameters, they are set according to the default values in our online experience address:
do_sample = false
temperature=0.9
top_k = 2

If you want to set do_sample = True
Then top_p = 0.8

**Reply by LeslieWongCV** [reply]
if do_sample=false, then no sampling when generate the token,
then will temperature still functional?

**Comment by FurkanGozukara** [comment]
@zRzRzRzRzRzRzR thank you so much

so if we set do sample the only parameter we add is top_p  = 0.8

top k and temperature remains same right? 

also cogagent-vqa is the best model for this right?

**Reply by zRzRzRzRzRzRzR** [reply]
yes, that is the default parameters

We don't have "the most suitable parameters". These are just our default parameters. You can debug them according to your scenario, but the default parameters are the most common.

**Comment by FurkanGozukara** [comment]
@zRzRzRzRzRzRzR thank you so much for replies

do you know why i might be getting these warnings?

![image](https://github.com/THUDM/CogVLM/assets/19240467/63ad0fd4-7414-4400-8772-e1e0911508b3)

**Reply by zRzRzRzRzRzRzR** [reply]
Are you using 4-bit to load? If so, this is normal, just ignore it.

**Reply by FurkanGozukara** [reply]
yes 4 bit thank you for fast reply
